slacker.go

```diff
@@ -130,36 +112,6 @@ func (s *Slacker) Interactive(interactiveEventHandler func(InteractiveBotContext
 	s.interactiveEventHandler = interactiveEventHandler
 }
 
-// CustomBotContext creates a new bot context
-func (s *Slacker) CustomBotContext(botContextConstructor func(context.Context, *slack.Client, *socketmode.Client, *MessageEvent) BotContext) {
-	s.botContextConstructor = botContextConstructor
-}
-
-// CustomInteractiveBotContext creates a new interactive bot context
-func (s *Slacker) CustomInteractiveBotContext(interactiveBotContextConstructor func(context.Context, *slack.Client, *socketmode.Client, *socketmode.Event) InteractiveBotContext) {
-	s.interactiveBotContextConstructor = interactiveBotContextConstructor
-}
-
-// CustomJobContext creates a new job context
-func (s *Slacker) CustomJobContext(jobContextConstructor func(context.Context, *slack.Client, *socketmode.Client) JobContext) {
-	s.jobContextConstructor = jobContextConstructor
-}
-
-// CustomCommand creates a new BotCommand
-func (s *Slacker) CustomCommand(commandConstructor func(usage string, definition *CommandDefinition) Command) {
-	s.commandConstructor = commandConstructor
-}
```

Comment:
This breaks https://github.com/shomali11/slacker/pull/91 / https://github.com/shomali11/slacker/pull/103. I don't think we have enough data as to how this is used to say if we provide another means to do what they want without it.